ThreeBotPackages/threefold/directory/actors/nodes.py

```diff
@@ -18,16 +32,16 @@ def add(self, node, schema_out=None):
 
         """
         # TODO check the sender is actually the node itself
-        # TODO: first check of the node already exist and if so do an update ?
-
-        # FIXME: we should not have to use _ddict since node is already a JSX object
-
+        if node.node_id:
```

Comment:
Trying to register a node with a node_id empty should raised an error.
Actually we should have some kind of validation steps here before we actually save the node into BCDB.